File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n=nums.size();
        int low=1,high=n-1;
        while(low<=high){
            int mid=(high+low)/2;
            int cnt=0;
            for(auto i:nums){
                if(i<=mid) cnt++;
            }
            if(cnt>mid){
                high=mid-1;
            }
            else low=mid+1;
        }
        return low;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        slow,fast=nums[0],nums[0]
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            if fast == slow:
                break
        slow1=nums[0]
        while slow1!=slow:
            slow1=nums[slow1]
            slow=nums[slow]
        return slow